### ISSUE TITLE
FileLogger cannot create log file with NLog or UWP

### ISSUE THREAD

**Issue by JKGMS** [NLog/NLog#2925]
Hi,
we have set up a simple test Project where we try out NLog. It works fine with Android but fails to create Logfiles when doing so in an UWP or NUnit Test project. iOS is untested at this point.
Basic idea: .net Standard 2.0 Library provides a platform independent configuration and a wrapper because we want to use a single instance of the logger instead of calling LogManager.GetCurrentClassLogger() everywhere. The Android, UWP etc. projects just provide a filepath for log files and create a logger instance. (Code snippets and internal logs are attached to the end of this issue)

**NLog version**: 4.5.10

**Current NLog config**
logFilePath is depending on platform:

- Android: (public folder of the App, works fine)
 `Application.Context.GetExternalFilesDir(null).AbsolutePath `

- UWP: (somewhere in AppData, doesn't work)
` ApplicationData.Current.LocalFolder.Path `

- NUnitTest: (just in %temp%, doesn't work)
 `Path.Combine(Path.GetTempPath(), "NLog");`

````
fileTarget = new FileTarget("target2")
{
	FileName = Path.Combine(logFilePath,"log.txt"),
	CreateDirs = true,
	KeepFileOpen = true,
	AutoFlush = true,
	ArchiveEvery = FileArchivePeriod.Day,
	MaxArchiveFiles = 5,
	ArchiveNumbering = ArchiveNumberingMode.Date, 
	ArchiveDateFormat = "yyyy-MM-dd",
	Layout = @"${date:format=HH\:mm\:ss} [${level}] ${message} ${exception}"
};
config.AddTarget(fileTarget);
config.AddRule(LogLevel.Debug, LogLevel.Fatal, fileTarget);
````

- What is the current result?
No logfile is created, only console logging works. The internal Logfiles contain different exceptions (please see attached logs).
- What is the expected result?
Log file should be created and filled in the given Paths.
- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
Yes, please see attachments
[logInternal_NUnit.txt](https://github.com/NLog/NLog/files/2401029/logInternal_NUnit.txt)
[logInternal_UWP.txt](https://github.com/NLog/NLog/files/2401030/logInternal_UWP.txt)
- Please post full exception details (message, stacktrace, inner exceptions)
See Attachments, both contain a single run of our NUnit or UWP Test to create and write into the logfile 
- Are there any workarounds? No/Unknown
- Is there a version in which it did work? No, we tried 4.5.9 and 4.5.10.
- Can you help us by writing an unit test?
I can provide the classes: (full project is sadly far beyond 10 MB)
     - Log=The wrapper
     - Logger=configuration
     - LoggerTests = Simple NUnit test
     [LoggingSnippets.zip](https://github.com/NLog/NLog/files/2401474/LoggingSnippets.zip)

If you need further info feel free to ask. Thanks.

**Comment by snakefoot** [comment]
The workaround for you is to configure both `KeepFileOpen=true` AND `concurrentWrites=false` . See also #2758

**Comment by snakefoot** [comment]
Created  #2926 to fix the issue with "Inheritance security rules violated by type"

**Comment by JKGMS** [comment]
Hi snakefoot, setting `concurrentWrites=false` helped with the UWP issue. Thanks for the quick response!

**Comment by snakefoot** [comment]
@JKGMS To work around the issue with NUnit then use the MemoryTarget (instead of FileTarget):

https://github.com/NLog/NLog/wiki/Memory-target

**Comment by 304NotModified** [comment]
PS: it's always bad practice to write to externals (files, database), in an unit test

**Comment by snakefoot** [comment]
@JKGMS NLog 4.5.11 has been released that fixes your issue with FileTarget in UnitTests: https://www.nuget.org/packages/NLog/

**Comment by 304NotModified** [comment]
Thanks @snakefoot . Will close this on as it's resolved with 4.5.11
